cmd/k3s/main.go

```diff
@@ -158,7 +158,20 @@ func stageAndRun(dataDir, cmd string, args []string) error {
 	}
 	logrus.Debugf("Asset dir %s", dir)
 
-	if err := os.Setenv("PATH", filepath.Join(dir, "bin")+":"+os.Getenv("PATH")+":"+filepath.Join(dir, "bin/aux")); err != nil {
```

Comment:
I think this just sets the path for the main `k3s` multicall wrapper, do we perhaps want to move the full patch logic somewhere else into the agent setup stuff perhaps?

Reply:
I don't think this is that big of a deal, who is directly calling the embedded K3s binary? I am not sure where this code should go, we need to set the PATH ASAP in the code sequence.

Reply:
yeah I guess it's fine, nothing should be running that directly. The only place that we bypass this wrapper is

A follow-up question - does this work with config.yaml. or only on the command-line? Should we support `1` in addition to `true` as the other bool flags like `--secrets-encryption` do?

Reply:
I have made a unit test around this and now handle all the edge cases with both config yaml values and the argument values. I am not allowing `=1` as a valid value, it must be `true` or `false` or "naked" (i.e. true).